wikiscraper.py

```diff
@@ -1,175 +1,264 @@
 #!/usr/bin/env python
 
+import argparse
+import copy
 import hashlib
 import json
 import mwparserfromhell
+import os.path
 import pprint
 import re
-import sys
-import urllib2
+import requests
 import yaml
 
-# http://stackoverflow.com/a/14692747
-def rget(dataDict, mapList):
-    return reduce(lambda d, k: d[k], mapList, dataDict)
+class WikiScraper:
+    """Used to scrape structured data from a set of MediaWiki pages"""
+
+    def __init__(self, configFileName, fromFiles):
+        """Set up the scraper"""
+        if not configFileName:
```

Comment:
General style comment: var names are generally underscored instead of camelCased.